pkg/resources/masking_policy.go

```diff
@@ -137,116 +134,124 @@ func MaskingPolicy() *schema.Resource {
 // CreateMaskingPolicy implements schema.CreateFunc.
 func CreateMaskingPolicy(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
 	name := d.Get("name").(string)
-	database := d.Get("database").(string)
-	schema := d.Get("schema").(string)
-	valueDataType := d.Get("value_data_type").(string)
-	maskingExpression := d.Get("masking_expression").(string)
+	databaseName := d.Get("database").(string)
+	schemaName := d.Get("schema").(string)
+
+	expression := d.Get("masking_expression").(string)
 	returnDataType := d.Get("return_data_type").(string)
 
-	builder := snowflake.MaskingPolicy(name, database, schema)
+	ctx := context.Background()
+	objectIdentifier := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+
+	signatureList := d.Get("signature").([]interface{})
+	signature := []sdk.TableColumnSignature{}
+	for _, s := range signatureList {
+		m := s.(map[string]interface{})
+		columns := m["column"].([]interface{})
+		for _, c := range columns {
+			cm := c.(map[string]interface{})
+			dt := sdk.DataTypeFromString(cm["type"].(string))
+			signature = append(signature, sdk.TableColumnSignature{
+				Name: cm["name"].(string),
+				Type: dt,
+			})
+		}
+	}
```

Comment:
no this is correct. when you declare a block you actually declare it as a set, which means there could be more than one block (except we set maximum to 1 and minimum to 1 so there isn't ever going to be anything else other than 1 in our case). since type is a set you have to first loop over it. then columns actually is a list, so that is correct too.

Reply:
Yes the second loop makes sense but should we be using a block for the signature then? There should always be exactly one so I don't think this is the right data structure.